src/pages/Login.js

```diff
@@ -44,17 +44,16 @@ function Login() {
       run(authApi.login(formData.email, formData.password));
     }
   };
-  localStorage.setItem('TOKEN', data?.token);
+
+  useEffect(() => localStorage.setItem('TOKEN', data?.token), [data?.token]);
 
   if (status === 'pending') {
     return <Spinner />;
   } else if (status === 'rejected') {
     throw error;
   } else if (status === 'resolved') {
     return (
-      <Route>
-        <AdminList />
-      </Route>
+      <Route>{auth.user ? <Redirect to="/adminList" /> : <Login />}</Route>
```

Comment:
here you're redirecting to login again, as you didn't trigger a request to get the profile from the token

* you're checking if there's a user and almost every time there's no user
* when you log in again, you'll get the profile for the previous token